src/bun.js/webcore/CompressionStreamEncoder.zig

```diff
@@ -0,0 +1,383 @@
+        .uninitialized => {
+            return globalObject.throwValue(globalObject.createErrorInstance("Encoder not initialized", .{}));
+        },
+    }
+
+    // Return accumulated output
+    if (this.pending_output.items.len > 0) {
+        // Transfer ownership of the buffer to JavaScript
+        const result = this.pending_output.toOwnedSlice() catch |err| {
+            return globalObject.throwError(err, "Out of memory");
+        };
+        return JSC.ArrayBuffer.fromBytes(result, .Uint8Array).toJS(globalObject);
+    }
+
+    return JSValue.jsNull();
+}
+
+pub fn flush(this: *CompressionStreamEncoder, globalObject: *JSGlobalObject, _: *JSC.CallFrame) bun.JSError!JSValue {
+    // Clear any pending output
+    this.pending_output.clearRetainingCapacity();
+
+    switch (this.state) {
+        .brotli => |encoder| {
+            // Flush remaining data with Brotli
+            const result = encoder.compressStream(.finish, "");
+            if (!result.success) {
+                return globalObject.throwValue(globalObject.createErrorInstance("Brotli flush failed", .{}));
+            }
+
+            // Add the initial output
+            if (result.output.len > 0) {
+                this.pending_output.appendSlice(result.output) catch |err| {
+                    return globalObject.throwError(err, "Out of memory");
+                };
+            }
+
+            // Brotli may have more output buffered - keep taking output until none left
+            while (encoder.hasMoreOutput()) {
+                const output_chunk = encoder.takeOutput();
+                if (output_chunk.len > 0) {
+                    this.pending_output.appendSlice(output_chunk) catch |err| {
+                        return globalObject.throwError(err, "Out of memory");
+                    };
+                }
+            }
+        },
+        .deflate, .gzip, .deflate_raw => |stream| {
+            // Flush remaining data with zlib
+            var output_buf: [16384]u8 = undefined;
+
+            stream.next_in = null;
+            stream.avail_in = 0;
+
+            while (true) {
+                stream.next_out = &output_buf;
+                stream.avail_out = output_buf.len;
+
+                const rc = zlib.deflate(stream, .Finish);
+
+                const bytes_written = output_buf.len - stream.avail_out;
+                if (bytes_written > 0) {
+                    this.pending_output.appendSlice(output_buf[0..bytes_written]) catch |err| {
+                        return globalObject.throwError(err, "Out of memory");
+                    };
+                }
+
+                if (rc == .StreamEnd) break;
+                if (rc != .Ok and rc != .BufError) {
+                    return globalObject.throwValue(globalObject.createErrorInstance("Flush failed", .{}));
+                }
+            }
+        },
+        .zstd => |stream| {
+            // Flush remaining data with zstd
+            var output_buf: [16384]u8 = undefined;
```

Comment:
May be too small . Each of these branches should live in a separate function to conserve stack space